File: Pensamiento Computacional en Práctica La Lógica Universal del Código a C++, Java, C#, JavaScript y Python/Codigo GO/algoritmos_capitulo4.go
package main

import (
	"fmt"
	"strconv"
)

// MenuPrincipal muestra el menú principal de la aplicación y permite navegar a los diferentes submódulos.
func MenuPrincipal() {
	var opcion int
	var salida bool
	datosmenuprincipal := []string{"Productos", "Punto de Venta", "Inventario", "Ventas", "Salida"}

	for !salida {
		opcionStr := DesplegarMenu("Menu de Punto de Tienda de Abarrotes la Pequeña", datosmenuprincipal)

		if opcionStr == "" {
			fmt.Println("opcion incorrecta")
		} else {
			opcion, _ = strconv.Atoi(opcionStr)
			switch opcion {
			case 1:
				MenuProductos()
			case 2:
				MenuPuntoVenta("000")
			case 3:
				MenuInventario()
			case 4:
				fmt.Println("Ventas")
			case 5:
				fmt.Println("Salida del Sistema")
				salida = true
			default:
				fmt.Println("No existe esta opcion")
			}
		}
	}
}

// MenuProductos muestra y gestiona las opciones relacionadas con los productos.
func MenuProductos() {
	datosmenuproductos := []string{"Modificar", "Listado", "Salida"}
	var opcion = "0"
	do {
		opcion = DesplegarMenu("Opciones de Productos", datosmenuproductos)
		if opcion == "" {
			fmt.Println("opcion incorrecta")
		} else {
			op, _ := strconv.Atoi(opcion)
			switch op {
			case 1:
				fmt.Println("Modificar Precio Producto")
			case 2:
				fmt.Println("Listado Producto")
			case 3:
				fmt.Println("Salida del Sistema")
			default:
				fmt.Println("No existe esta opcion")
			}
		}
	} while opcion != "3"
}

// MenuPuntoVenta gestiona el menú del punto de venta para un ticket específico.
func MenuPuntoVenta(idticket string) {
	var opcion, membrete string
	pago := false
	idticket = IdTicketSiguiente(idticket)
	fechadia := Fecha()
	membrete = "Fecha del Dia " + fechadia + " Ticket No " + idticket
	membrete += "
-----------------------------------------------------
"
	datosmenu := []string{"Agregar", "Eliminar", "Listado", "Pagar", "Salida"}

	do {
		opcion = DesplegarMenu(membrete+"
Menu de Punto de Venta", datosmenu)
		if opcion == "" {
			fmt.Println("dato incorrecto introducido")
		} else {
			op, _ := strconv.Atoi(opcion)
			switch op {
			case 1:
				fmt.Println("Agregar Articulo al Ticket")
			case 2:
				fmt.Println("Eliminar Articulo del Ticket")
			case 3:
				fmt.Println("Listado de Articulos del Ticket")
			case 4:
				fmt.Println("Pagar Ticket y salir")
				opcion = "5"
				pago = true
			case 5:
				fmt.Println("Salida del Ventas")
				if !pago {
					fmt.Println("No pago el ticket")
					fmt.Println("eliminando ticket", idticket)
				}
			default:
				fmt.Println("No existe esta opcion")
			}
		}
	} while opcion != "5"
}

// MenuInventario muestra y gestiona las opciones del inventario.
func MenuInventario() {
	datosmenuinventario := []string{"Listado", "Agregar", "Salida"}
	var opcion = "0"
	do {
		opcion = DesplegarMenu("Opciones de Inventarios", datosmenuinventario)
		if opcion == "" {
			fmt.Println("opcion incorrecta")
		} else {
			op, _ := strconv.Atoi(opcion)
			switch op {
			case 1:
				fmt.Println("Listado de Inventario")
			case 2:
				fmt.Println("Agregar Inventario")
			case 3:
				fmt.Println("Salida del Sistema")
			default:
				fmt.Println("No existe esta opcion")
			}
		}
	} while opcion != "3"
}

File: Pensamiento Computacional en Práctica La Lógica Universal del Código a C++, Java, C#, JavaScript y Python/Codigo RUST/algoritmos_capitulo4.rs
use std::io;

fn menu_principal() {
    let mut opcion: i32;
    let mut salida = false;
    let datosmenuprincipal = vec!["Productos", "Punto de Venta", "Inventario", "Ventas", "Salida"];

    while !salida {
        let opcion_str = desplegar_menu("Menu de Punto de Tienda de Abarrotes la Pequeña", &datosmenuprincipal);

        if opcion_str.is_empty() {
            println!("opcion incorrecta");
        } else {
            opcion = opcion_str.parse().unwrap_or(0);
            match opcion {
                1 => menu_productos(),
                2 => menu_punto_venta("000"),
                3 => menu_inventario(),
                4 => println!("Ventas"),
                5 => {
                    println!("Salida del Sistema");
                    salida = true;
                }
                _ => println!("No existe esta opcion"),
            }
        }
    }
}

fn menu_productos() {
    let datosmenuproductos = vec!["Modificar", "Listado", "Salida"];
    let mut opcion = "0".to_string();
    while opcion != "3" {
        opcion = desplegar_menu("Opciones de Productos", &datosmenuproductos);
        if opcion.is_empty() {
            println!("opcion incorrecta");
        } else {
            let op: i32 = opcion.parse().unwrap_or(0);
            match op {
                1 => println!("Modificar Precio Producto"),
                2 => println!("Listado Producto"),
                3 => println!("Salida del Sistema"),
                _ => println!("No existe esta opcion"),
            }
        }
    }
}

fn menu_punto_venta(idticket: &str) {
    let mut opcion: String;
    let mut pago = false;
    let mut idticket = id_ticket_siguiente(idticket);
    let fechadia = fecha();
    let mut membrete = format!("Fecha del Dia {} Ticket No {}\n-----------------------------------------------------\n", fechadia, idticket);
    let datosmenu = vec!["Agregar", "Eliminar", "Listado", "Pagar", "Salida"];

    loop {
        opcion = desplegar_menu(&format!("{}\nMenu de Punto de Venta", membrete), &datosmenu);
        if opcion.is_empty() {
            println!("dato incorrecto introducido");
        } else {
            let op: i32 = opcion.parse().unwrap_or(0);
            match op {
                1 => println!("Agregar Articulo al Ticket"),
                2 => println!("Eliminar Articulo del Ticket"),
                3 => println!("Listado de Articulos del Ticket"),
                4 => {
                    println!("Pagar Ticket y salir");
                    opcion = "5".to_string();
                    pago = true;
                }
                5 => {
                    println!("Salida del Ventas");
                    if !pago {
                        println!("No pago el ticket");
                        println!("eliminando ticket {}", idticket);
                    }
                    break;
                }
                _ => println!("No existe esta opcion"),
            }
        }
    }
}

fn menu_inventario() {
    let datosmenuinventario = vec!["Listado", "Agregar", "Salida"];
    let mut opcion = "0".to_string();
    while opcion != "3" {
        opcion = desplegar_menu("Opciones de Inventarios", &datosmenuinventario);
        if opcion.is_empty() {
            println!("opcion incorrecta");
        } else {
            let op: i32 = opcion.parse().unwrap_or(0);
            match op {
                1 => println!("Listado de Inventario"),
                2 => println!("Agregar Inventario"),
                3 => println!("Salida del Sistema"),
                _ => println!("No existe esta opcion"),
            }
        }
    }
}
